tardis/energy_input/gamma_ray_channel.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import numpy as np
+import pandas as pd
+import astropy.units as u
+import radioactivedecay as rd
+
+
+from tardis.energy_input.energy_source import (
+    get_nuclear_lines_database,
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def create_isotope_dicts(raw_isotope_abundance, cell_masses):
+    """
+    Function to create a dictionary of isotopes for each shell with their masses.
+
+    Parameters
+    ----------
+    raw_isotope_abundance : pd.DataFrame
+        isotope abundance in mass fractions.
+    cell_masses : numpy.ndarray
+        shell masses in units of g
+
+    Returns
+    -------
+        isotope_dicts : Dict
+            dictionary of isotopes for each shell with their ``masses``.
+            For eg: {0: {(28, 56): {'Ni56': 0.0001}, (27, 57): {'Co56': 0.0001}}
```

Comment:
I don't think this is actually how the dict looks based on the code

Reply:
I think it is more like `{0: {"Ni56": abundance * mass}}`

Reply:
Changed.